caluma/caluma_form/models.py

```diff
@@ -245,12 +285,20 @@ class Meta:
 class File(UUIDModel):
     name = models.CharField(max_length=255)
 
+    def _get_file_names(self):
+        old_file = self.history.first()
+        new_name = f"{old_file.pk}_{old_file.name}"
+        return (self.object_name, new_name)
+
     def _move_blob(self):
         # move the file on update
         # this makes sure it stays available when querying the history
-        old_file = self.history.first()
-        new_name = f"{old_file.pk}_{old_file.name}"
-        client.move_object(self.object_name, new_name)
+        client.move_object(*self._get_file_names())
+
+    def copy(self, *args, **kwargs):
+        old_name, new_name = self._get_file_names()
```

Comment:
Correct me if I'm wrong: This constructs the new file name from the pk of the newest historical entry. Why?

Reply:
Addressing the top question as well:
You're right, something is off.
I think we need to create and copy of `File`, because we need to know it's pk to copy the blob to the right name. And return the copy ofc, so the name `copy()` would still make sense.

Reply:
The logic should be: Create a new `File` object with the same name as the original one. And then call

```python
client.copy_object(original.object_name, new_file.object_name)
```

Reply:
Yes that's what I just did. :)
Though the name of the function would stay `copy()`.

Reply:
Now the name `copy()` makes sense of course.